libopendroneid/wifi.c

```diff
@@ -0,0 +1,340 @@
+	mprintf("\t\t\t\"UASID\": %s\n", UAS_Data->BasicID.UASID);
+	mprintf("\t\t},\n");
+
+	mprintf("\t\t\"Location\": {\n");
+	mprintf("\t\t\t\"Status\": %d,\n", (int)UAS_Data->Location.Status);
+	mprintf("\t\t\t\"Direction\": %f,\n", UAS_Data->Location.Direction);
+	mprintf("\t\t\t\"SpeedHorizontal\": %f,\n", UAS_Data->Location.SpeedHorizontal);
+	mprintf("\t\t\t\"SpeedVertical\": %f,\n", UAS_Data->Location.SpeedVertical);
+	mprintf("\t\t\t\"Latitude\": %f,\n", UAS_Data->Location.Latitude);
+	mprintf("\t\t\t\"Longitude\": %f,\n", UAS_Data->Location.Longitude);
+	mprintf("\t\t\t\"AltitudeBaro\": %f,\n", UAS_Data->Location.AltitudeBaro);
+	mprintf("\t\t\t\"AltitudeGeo\": %f,\n", UAS_Data->Location.AltitudeGeo);
+	mprintf("\t\t\t\"Height\": %f,\n", UAS_Data->Location.Height);
+	mprintf("\t\t\t\"HorizAccuracy\": %d,\n", UAS_Data->Location.HorizAccuracy);
+	mprintf("\t\t\t\"VertAccuracy\": %d,\n", UAS_Data->Location.VertAccuracy);
+	mprintf("\t\t\t\"SpeedAccuracy\": %d,\n", UAS_Data->Location.SpeedAccuracy);
+	mprintf("\t\t\t\"TSAccuracy\": %d,\n", UAS_Data->Location.TSAccuracy);
+	mprintf("\t\t\t\"TimeStamp\": %f\n", UAS_Data->Location.TimeStamp);
+	mprintf("\t\t},\n");
+
+	mprintf("\t\t\"Authentication\": {\n");
+	mprintf("\t\t\t\"AuthType\": %i,\n", UAS_Data->Auth[0].AuthType);
+	mprintf("\t\t\t\"AuthToken\": %s\n", UAS_Data->Auth[0].AuthData);
+	mprintf("\t\t},\n");
+
+	mprintf("\t\t\"SelfID\": {\n");
+	mprintf("\t\t\t\"Name\": \"string\",\n");
+	mprintf("\t\t\t\"Description\": %s\n", UAS_Data->SelfID.Desc);
+	mprintf("\t\t},\n");
+
+	mprintf("\t\t\"Operator\": {\n");
+	mprintf("\t\t\t\"LocationSource\": %i,\n", UAS_Data->System.LocationSource);
+	mprintf("\t\t\t\"OperatorLatitude\": %f,\n", UAS_Data->System.OperatorLatitude);
+	mprintf("\t\t\t\"OperatorLongitude\": %f,\n", UAS_Data->System.OperatorLongitude);
+	mprintf("\t\t\t\"AreaCount\": %i,\n", UAS_Data->System.AreaCount);
+	mprintf("\t\t\t\"AreaRadius\": %i,\n", UAS_Data->System.AreaRadius);
+	mprintf("\t\t\t\"AreaCeiling\": %f\n", UAS_Data->System.AreaCeiling);
+	mprintf("\t\t}\n");
+
+	mprintf("\t}\n}");
+
+	return drone_str;
+}
+
+int odid_message_encode_pack(ODID_UAS_Data *UAS_Data, void *pack, size_t buflen)
+{
+	ODID_MessagePack_encoded *outPack;
+	size_t len = 0;
+
+	/* check if there is enough space for the header. */
+	if (sizeof(*outPack) > buflen)
+		return -ENOMEM;
+
+	/* TODO: flexibly set optional fields as available */
+
+	outPack = (ODID_MessagePack_encoded *) pack;
+	outPack->ProtoVersion = 0;
+	outPack->MessageType = 0;
+	outPack->SingleMessageSize = ODID_MESSAGE_SIZE;
+	outPack->MsgPackSize = 5;
+	len += sizeof(*outPack);
+
+	if (len + (outPack->MsgPackSize * ODID_MESSAGE_SIZE) > buflen)
+		return -ENOMEM;
+
+	encodeBasicIDMessage((void *)&outPack->Messages[0], &UAS_Data->BasicID);
+	encodeLocationMessage((void *)&outPack->Messages[1], &UAS_Data->Location);
+	encodeAuthMessage((void *)&outPack->Messages[2], &UAS_Data->Auth[0]);
```

Comment:
Additional auth messages and also the operator ID message is being ignored here.